copilot/fastapi/app/tvp.py

```
@@ -0,0 +1,392 @@
+        repo_url (str): URL of the Git repository to clone.
+        repo_path (str): Local path where the repository should be cloned.
+        branch (str): Branch to check out.
+        
+    Raises:
+        subprocess.CalledProcessError: If Git clone operation fails.
+        OSError: If directory creation fails.
+    """
+    os.makedirs(os.path.dirname(repo_path), exist_ok=True)
+    try:
+        # Add timeout to prevent hanging on network issues
+        subprocess.run(
```

Ruff (0.8.2)

`subprocess` call: check for execution of untrusted input

(S603)